Lib/ast.py

```diff
@@ -565,10 +606,16 @@ class Ellipsis(Constant, metaclass=_ABC):
     _fields = ()
 
     def __new__(cls, *args, **kwargs):
-        if cls is Ellipsis:
+        if cls is _ast_Ellipsis:
+            import warnings
+            warnings._deprecated(
+                f"ast.Ellipsis", message=_DEPRECATED_CLASS_MESSAGE, remove=(3, 14)
+            )
             return Constant(..., *args, **kwargs)
         return Constant.__new__(cls, *args, **kwargs)
 
+_ast_Ellipsis = Ellipsis
```

Comment:
IIUC, this is defined as an alias to avoid an additional deprecation warning from `__getattr__()` below. If I've got that right, is it worth adding a comment to that effect?

Reply:
Not quite! It took me a moment to understand this as well when I took this PR over from Serhiy, so I should add a comment.

The issue is that we need to do an `if cls is Ellipsis` check here: https://github.com/python/cpython/pull/104199/files#diff-abeeed217a24ce3c21e15c2e32e2f0886cea69986c2641455c1eefc983c1f977R609

But we remove `Ellipsis` from the global namespace here (note that the comprehension does `globals().pop()`, so that we can do the module-level `__getattr__` thing: https://github.com/python/cpython/pull/104199/files#diff-abeeed217a24ce3c21e15c2e32e2f0886cea69986c2641455c1eefc983c1f977R1750

So if we didn't add the alias here, we wouldn't have an easy-to-access reference to the class in the global namespace that we cls use for the `is` check in `Ellipsis.__new__`!

Reply:
Right! Thanks for the explanation.

Reply:
Added a comment in 915e12e